// @elevenlabs/elevenlabs-js/dist/api/resources/workspace/client/Client.js
"use strict";
// This file was auto-generated by Fern from our API Definition.
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkspaceClient = void 0;
const BaseClient_1 = require("../../../../BaseClient");
const Client_1 = require("../resources/auditLogs/client/Client");
const Client_2 = require("../resources/authConnections/client/Client");
const Client_3 = require("../resources/groups/client/Client");
const Client_4 = require("../resources/invites/client/Client");
const Client_5 = require("../resources/members/client/Client");
const Client_6 = require("../resources/resources/client/Client");
const Client_7 = require("../resources/usage/client/Client");
class WorkspaceClient {
    constructor(options = {}) {
        this._options = (0, BaseClient_1.normalizeClientOptions)(options);
    }
    get auditLogs() {
        var _a;
        return ((_a = this._auditLogs) !== null && _a !== void 0 ? _a : (this._auditLogs = new Client_1.AuditLogsClient(this._options)));
    }
    get authConnections() {
        var _a;
        return ((_a = this._authConnections) !== null && _a !== void 0 ? _a : (this._authConnections = new Client_2.AuthConnectionsClient(this._options)));
    }
    get groups() {
        var _a;
        return ((_a = this._groups) !== null && _a !== void 0 ? _a : (this._groups = new Client_3.GroupsClient(this._options)));
    }
    get invites() {
        var _a;
        return ((_a = this._invites) !== null && _a !== void 0 ? _a : (this._invites = new Client_4.InvitesClient(this._options)));
    }
    get members() {
        var _a;
        return ((_a = this._members) !== null && _a !== void 0 ? _a : (this._members = new Client_5.MembersClient(this._options)));
    }
    get resources() {
        var _a;
        return ((_a = this._resources) !== null && _a !== void 0 ? _a : (this._resources = new Client_6.ResourcesClient(this._options)));
    }
    get usage() {
        var _a;
        return ((_a = this._usage) !== null && _a !== void 0 ? _a : (this._usage = new Client_7.UsageClient(this._options)));
    }
}
exports.WorkspaceClient = WorkspaceClient;
